src/bitcoin-tx.cpp

```diff
@@ -298,19 +308,30 @@ static void MutateTxAddOutData(CMutableTransaction &tx, const string &strInput)
 static void MutateTxAddOutScript(CMutableTransaction &tx, const string &strInput)
 {
     // separate VALUE:SCRIPT in string
-    size_t pos = strInput.find(':');
-    if ((pos == string::npos) || (pos == 0))
+    std::vector<std::string> vStrInputParts;
+    boost::split(vStrInputParts, strInput, boost::is_any_of(":"));
+    if (vStrInputParts.size() < 2)
         throw runtime_error("TX output missing separator");
 
     // extract and validate VALUE
-    string strValue = strInput.substr(0, pos);
-    CAmount value;
-    if (!ParseMoney(strValue, value))
-        throw runtime_error("invalid TX output value");
+    CAmount value = ExtractAndValidateValue(vStrInputParts[0]);
 
     // extract and validate script
-    string strScript = strInput.substr(pos + 1, string::npos);
-    CScript scriptPubKey = ParseScript(strScript); // throws on err
+    std::string strScript = vStrInputParts[1];
+    CScript scriptPubKey = ParseScript(strScript);
+
+    // Extract FLAGS
+    bool bScriptHash = false;
+    if (vStrInputParts.size() == 3)
```

Comment:
If the string has 4 parts this code will ignore the flags.  To maintain compatibility with the previous behavior, we would ignore additional sections.  so this should be: if (vStrInputParts.size() > 2)

Reply:
fixed